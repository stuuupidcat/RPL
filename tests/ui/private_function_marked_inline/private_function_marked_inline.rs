#[inline]
fn foo11() {}
//~^ERROR: it usually isn't necessary to apply #[inline] to private functions

#[inline]
pub(crate) fn foo12() {}
//~^ERROR: it usually isn't necessary to apply #[inline] to private functions
#[inline]
pub fn foo13() {}

#[inline(always)]
fn foo21() {}
//~^ERROR: it usually isn't necessary to apply #[inline] to private functions

#[inline(always)]
pub(crate) fn foo22() {}
//~^ERROR: it usually isn't necessary to apply #[inline] to private functions

#[inline(always)]
pub fn foo23() {}

#[inline(never)]
fn foo31() {}

#[inline(never)]
pub(crate) fn foo32() {}

#[inline(never)]
pub fn foo33() {}

trait Trait {
    #[inline]
    fn foo_in_trait() {}
    //~^ERROR: it usually isn't necessary to apply #[inline] to private functions
    //~|ERROR: it usually isn't necessary to apply #[inline] to generic functions
}

struct Struct;

impl Struct {
    #[inline]
    fn foo_in_impl() {}
    //~^ERROR: it usually isn't necessary to apply #[inline] to private functions
}

macro_rules! private_inline {
    ($ident:ident) => {
        #[inline]
        fn $ident() {}
        //~^ERROR: it usually isn't necessary to apply #[inline] to private functions
    };
    (#[$meta:meta] $ident:ident) => {
        #[$meta]
        fn $ident() {}
        //~^ERROR: it usually isn't necessary to apply #[inline] to private functions
    };
}

private_inline!(bar1);
private_inline!(#[inline] bar2);

fn main() {}
