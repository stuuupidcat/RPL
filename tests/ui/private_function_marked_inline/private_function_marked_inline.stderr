error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:2:1
   |
LL | #[inline]
   | --------- `#[inline]` here
LL | fn foo11() {}
   | ^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions
   = note: `-D rpl::private-function-marked-inline` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(rpl::private_function_marked_inline)]`

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:6:1
   |
LL | #[inline]
   | --------- `#[inline]` here
LL | pub(crate) fn foo12() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:12:1
   |
LL | #[inline(always)]
   | ----------------- `#[inline]` here
LL | fn foo21() {}
   | ^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:16:1
   |
LL | #[inline(always)]
   | ----------------- `#[inline]` here
LL | pub(crate) fn foo22() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:33:5
   |
LL |     #[inline]
   |     --------- `#[inline]` here
LL |     fn foo_in_trait() {}
   |     ^^^^^^^^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions

error: it usually isn't necessary to apply #[inline] to generic functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:33:5
   |
LL |     #[inline]
   |     --------- `#[inline]` here
LL |     fn foo_in_trait() {}
   |     ^^^^^^^^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html and https://rustc-dev-guide.rust-lang.org/backend/monomorph.html
   = note: generic functions are always `#[inline]` (monomorphization)
   = note: `-D rpl::generic-function-marked-inline` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(rpl::generic_function_marked_inline)]`

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:42:5
   |
LL |     #[inline]
   |     --------- `#[inline]` here
LL |     fn foo_in_impl() {}
   |     ^^^^^^^^^^^^^^^^^^^ `#[inline]` applied here
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:49:9
   |
LL |         #[inline]
   |         --------- `#[inline]` here
LL |         fn $ident() {}
   |         ^^^^^^^^^^^^^^ `#[inline]` applied here
...
LL | private_inline!(bar1);
   | --------------------- in this macro invocation
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions
   = note: this error originates in the macro `private_inline` (in Nightly builds, run with -Z macro-backtrace for more info)

error: it usually isn't necessary to apply #[inline] to private functions
  --> tests/ui/private_function_marked_inline/private_function_marked_inline.rs:54:9
   |
LL |         #[$meta]
   |         -------- `#[inline]` here
LL |         fn $ident() {}
   |         ^^^^^^^^^^^^^^ `#[inline]` applied here
...
LL | private_inline!(#[inline] bar2);
   | ------------------------------- in this macro invocation
   |
   = help: See https://matklad.github.io/2021/07/09/inline-in-rust.html
   = note: the compiler generally makes good inline decisions about private functions
   = note: this error originates in the macro `private_inline` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 9 previous errors

